students/JoeNunnelley/session07/html_render.py

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/env python3
+
+"""
+A class-based system for rendering html.
+"""
+# This is the framework for the base class
+class Element():
+    """
+    An element has the format
+        <tag [attribute=value, attribute=value /]>[text</tag>]
+        lead tags can have attributes
+        lead tags can exist by themselves but must end in a /
+        blocks that have text must have a closing tag of the form </tag>
+        tag blocks can have x children and be nested to any depth
+    """
+    tag = 'html'
+    write_doctype = False
+    indent = '   '
+    depth = 0
+
+
+    def __init__(self, content=None):
+        """
+        Initialize the class variables
+        """
+        self.attributes = []
+        self.text = []
+        self.children = []
+        self.append(content)
+
+    def get(self):
+        """
+        Base function to get block as a string
+        """
+        pass
+
+    def append(self, new_content):
+        """
+        Append content to an element.
+        There are four methods of doing this:
+        1.) Using a class that is derived from Element
+        2.) Using a dictionary containing iterable entries for attributes, text and children
+        3.) Using a string to set just the text value of the tag
+        4.) Using an empty new_content to just create the object with no values
+        """
+        if new_content is not None and self.tag is not None:
+            if isinstance(new_content, Element):
+                if isinstance(new_content, A):
+                    self.text.append(new_content.get())
+                else:
+                    new_content.depth = self.depth + 1
+                    self.children.append(new_content)
+            elif isinstance(new_content, dict):
+                self.attributes.append(new_content['attributes'])
+                self.text.append(new_content['text'])
+                self.children = new_content['children']
+            elif isinstance(new_content, str):
+                self.text.append(new_content)
+            elif new_content is None:
```

Comment:
I don't believe you need this since it will never enter the if statement if new_content is None